src/Microsoft.OData.Core/UriParser/Visitors/IsCollectionTranslator.cs

```diff
@@ -98,6 +98,42 @@ public override bool Translate(CountSegment segment)
             return false;
         }
 
+        /// <summary>
+        /// Translate a FilterSegment
+        /// </summary>
+        /// <param name="segment">the segment to Translate</param>
+        /// <returns>UserDefinedValue</returns>
+        /// <exception cref="System.ArgumentNullException">Throws if the input segment is null.</exception>
+        public override bool Translate(FilterSegment segment)
+        {
+            ExceptionUtils.CheckArgumentNotNull(segment, "segment");
+            return true;
+        }
+
+        /// <summary>
+        /// Translate a ReferenceSegment
+        /// </summary>
+        /// <param name="segment">the segment to Translate</param>
+        /// <returns>UserDefinedValue</returns>
+        /// <exception cref="System.ArgumentNullException">Throws if the input segment is null.</exception>
+        public override bool Translate(ReferenceSegment segment)
+        {
+            ExceptionUtils.CheckArgumentNotNull(segment, "segment");
+            return false;

```

Comment:
>return false; [](start = 12, length = 13)

Wouldn't this depend on whether the reference segment was applied to a collection (i.e., Employees(1)/ns.Manager/Employees/$ref) or a single value (i.e., Employees(1)/Manager/$ref)?

Reply:
Changed to return `!segment.SingleResult;`